web/modules/custom/dpl_reservations/src/Plugin/Block/ReservationListBlock.php

```diff
@@ -0,0 +1,200 @@
+<?php
+
+namespace Drupal\dpl_reservations\Plugin\Block;
+
+use Drupal\Core\Block\BlockBase;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\dpl_react_apps\Controller\DplReactAppsController;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\dpl_library_agency\Branch\BranchRepositoryInterface;
+use Drupal\dpl_library_agency\BranchSettings;
+
+/**
+ * Provides user reservations list.
+ *
+ * @Block(
+ *   id = "dpl_reservations_list_block",
+ *   admin_label = "List user reservations"
+ * )
+ */
+class ReservationListBlock extends BlockBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * Drupal config factory.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  private ConfigFactoryInterface $configFactory;
+
+  /**
+   * ReservationListBlock constructor.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin ID for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $configFactory
+   *   Drupal config factory to get FBS and Publizon settings.
+   * @param \Drupal\dpl_library_agency\BranchSettings $branchSettings
+   *   The branchsettings for branch config.
+   * @param \Drupal\dpl_library_agency\Branch\BranchRepositoryInterface $branchRepository
+   *   The branchsettings for getting branches.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, ConfigFactoryInterface $configFactory, protected BranchSettings $branchSettings, protected BranchRepositoryInterface $branchRepository) {
```

Comment:
Please manage the PHPStan warnings.

If you do not think it is valuable to add types to iterables here then one option might be to ignore this for certain block functions.

Reply:
Fixed